### ISSUE TITLE
Move from Lerna to alternative toolset (like Nx)

### ISSUE THREAD

**Issue by seyoon20087** [facebook/docusaurus#7333]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Motivation

[Lerna is not maintained anymore](https://github.com/lerna/lerna/issues/2703). As a result, you may want to switch to an alternative solution (such as [Nx](https://nx.dev/migration/adding-to-monorepo)).

### Self-service

- [ ] I'd be willing to do some initial work on this proposal myself.

**Comment by Josh-Cena** [comment]
Agree 👍 I've also talked to @slorber about this on Discord. My main candidate for now is monodeploy, because we are also migrating to Yarn v3 in #6028. cc @RDIL in case you want to make the DX fully prepared in that PR. The publishing process needs investigation.

**Comment by slorber** [comment]
We also have good feedback about using Turborepo from @yangshun 

Please don't submit any PR on this for now, we'll figure out the setup we want to adopt

For the time being, our Lerna setup still works

**Comment by Josh-Cena** [comment]
https://github.com/lerna/lerna/issues/3121 Large thonk

**Comment by yangshun** [comment]
If we move to Vercel hosting then using Turborepo would be a no-brainer.

**Comment by Josh-Cena** [comment]
Does Vercel hosting offer special integration with TR? If that, I'm very interested in seeing a working example 😄

**Comment by yangshun** [comment]
Yes, they have [remote caching](https://turborepo.org/docs/core-concepts/remote-caching), but I'm not sure whether it's free or a paid feature.

**Comment by Josh-Cena** [comment]
A great amount of our CI time (in PR) is spent on building packages. If we can cut that down (through persisting some cache), we can greatly speed up the deploy preview. Looking forward to any solution.

**Comment by slorber** [comment]
Afaik Meta is already using Vercel for many sites (like the new Beta site) and maybe not using Nx cloud already. 

I'd rather ask @zpao what is possible for us to use there before choosing the definitive tool. 

Having caching would be very nice for sure.

Lerna still works and is not maintained again, so there's no need to hurry

**Comment by Josh-Cena** [comment]
Lerna 5.0 has just been released, with a lot of dependency upgrades! This makes me hate it a lot less.

**Comment by pavankjadda** [comment]
> Lerna 5.0 has just been released, with a lot of dependency upgrades! This makes me hate it a lot less.

It seems Lerna 5.1.x is way faster than TR (5.3 times). Quoting Victor [tweet](https://twitter.com/victorsavkin/status/1529493314621145090?s=20&t=N6RR9eTNSw8agItpisPkKw)
> 🎉 Lerna now supports:
>- distributed caching
>- tasks pipelines (e.g., tests can depend on builds)
>- distributed task execution

>It's super fast 🚀

>➡️ Lerna is 5.3 times faster than Turborepo!

> Blog: [link](https://blog.nrwl.io/lerna-used-to-walk-now-it-can-fly-eab7a0fe7700?source=friends_link&sk=6c827ec7c9adfc1c760ff2e3f3e05cc7)

**Comment by Josh-Cena** [comment]
I'm not even sure if we need to do anything at this stage, since I don't know if it's Lerna or tsc being slow, but I'd still like to play with different managers just to confirm. I'd definitely try out 5.1 in this weekend's dependency upgrade.

**Comment by RDIL** [comment]
@Josh-Cena usually the performance boils down to:

- TypeScript's type checking portion is always on, and the more types it finds (so essentially every type def in node modules), the slower it is.
- Lerna is mostly just Git subprocess operations or startup, both of which are a bit slow.

**Comment by maverictopgun72** [comment]
https://github.com/facebook/docusaurus/issues/7333#issue-1226196895
